pkg/controller/plan/migration.go

```diff
@@ -1631,20 +1632,47 @@ func (r *Migration) updatePopulatorCopyProgress(vm *plan.VMStatus, step *plan.St
 		}
 
 		percent := float64(transferredBytes/0x100000) / float64(task.Progress.Total)
-		task.Progress.Completed = int64(percent * float64(task.Progress.Total))
+		newProgress := int64(percent * float64(task.Progress.Total))
+		err = r.checkPopulatorPod(newProgress, task.Progress.Completed, string(pvc.UID))
+		if err != nil {
+			return
+		}
+		task.Progress.Completed = newProgress
 	}
 
 	step.ReflectTasks()
 	return
 }
 
+// Checks if the populator pod failed when the progress didn't change
+func (r *Migration) checkPopulatorPod(newProgress, oldProgress int64, givenPvcId string) error {
+	if newProgress != oldProgress {
```

Comment:
it's a bit weird to pass the progress to this function just to check that they are different here, how about doing that in updatePopulatorCopyProgress?